src/instructlab/config/init.py

```diff
@@ -194,6 +194,12 @@ def prompt_user_to_choose_vendors(
         click.echo(
             "No profile selected - ilab will use generic code defaults - these may not be optimized for your system."
         )
+    else:
+        click.secho(
```

Comment:
Would it be better to re-ask rather than exit?

Reply:
That is a good suggestion.  Should that be part of another PR since that UX change to retry would need to be implemented in the vendor and hardware system selection.

Is there any requirement to fix this quickly for a `v0.21.z-stream` release and, if so, I can rebase this on top of `release-v0.21` branch then we can do another issues/PR for a retry option?  As it stands now the v0.21.z releases will reset the system profile config without warning if the user accidentally hits Enter or inputs an invalid number

Reply:
No objection if we need to do a quick fix now and update later.  Re: 

> retry would need to be implemented in the vendor and hardware system selection

Not quite following this statement.  Wouldn't the loop be around the `click.prompt` just above?

Reply:
`click.IntRange` did the trick